File: monitor-core/src/main/java/monitorCore/Model/Animal.java
package monitorCore.Model;

import lombok.*;

import javax.persistence.Entity;
import java.io.Serializable;


@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Builder
@Entity(name="Animal")
public class Animal extends BaseEntity implements Serializable {
    private String breed;
    private String name;
    private int age;
    private String type;
    private int points;
}

File: src/app/Model/Animal.ts
import {Entity} from './Entity';

export class Animal extends Entity{
  breed: string;
  name: string;
  age: number;
  type: string;
  points: number;
  constructor(id ?: number, name ?: string, breed ?: string, age ?: number, type ?: string, points ?: number) {
    super(id);
    this.breed = breed;
    this.name = name;
    this.age = age;
    this.type = type;
    this.points = points;
  }
}
